statsmodels/discrete/count_model.py

```diff
@@ -0,0 +1,721 @@
+import statsmodels.regression.linear_model as lm
+from statsmodels.discrete.discrete_model import (DiscreteModel, CountModel,
+                                                 Poisson, Logit, CountResults,
+                                                 L1CountResults, Probit,
+                                                 _discrete_results_docs,
+                                                 GeneralizedPoisson,
+                                                 NegativeBinomialP)
+from statsmodels.distributions import zipoisson, zigenpoisson, zinegbin
+from statsmodels.tools.numdiff import (approx_fprime, approx_hess,
+                                       approx_hess_cs, approx_fprime_cs)
+from statsmodels.tools.decorators import (resettable_cache, cache_readonly)
+
+
+class GenericZeroInflated(CountModel):
+    __doc__ = """
+    Generiz Zero Inflated model for count data
+
+    %(params)s
+    %(extra_params)s
+
+    Attributes
+    -----------
+    endog : array
+        A reference to the endogenous response variable
+    exog : array
+        A reference to the exogenous design.
+    exog_infl: array
+        A reference to the zero-inflated exogenous design.
+    """ % {'params' : base._model_params_doc,
+           'extra_params' :
+           """offset : array_like
+        Offset is added to the linear prediction with coefficient equal to 1.
+    exposure : array_like
+        Log(exposure) is added to the linear prediction with coefficient
+        equal to 1.
+
+    """ + base._missing_param_doc}
+
+    def __init__(self, endog, exog, exog_infl=None, offset=None,
+                 inflation='logit', exposure=None, missing='none', **kwargs):
+        super(GenericZeroInflated, self).__init__(endog, exog, offset=offset,
+                                                  exposure=exposure,
+                                                  missing=missing, **kwargs)
+
+        if exog_infl is None:
+            self.k_inflate = 1
+            self.exog_infl = np.ones((endog.size, self.k_inflate))
+        else:
+            self.exog_infl = exog_infl
+            self.k_inflate = exog_infl.shape[1]
+
+        if len(exog.shape) == 1:
+            self.k_exog = 1
+        else:
+            self.k_exog = exog.shape[1]
+
+        self.infl = inflation
+        if inflation == 'logit':
+            self.model_infl = Logit(np.zeros(self.exog_infl.shape[0]),
+                                    self.exog_infl)
+            self._hessian_inflate = self._hessian_logit
+        elif inflation == 'probit':
+            self.model_infl = Probit(np.zeros(self.exog_infl.shape[0]),
+                                    self.exog_infl)
+            self._hessian_inflate = self._hessian_probit
+
+        else:
+            raise TypeError("inflation == %s, which is not handled"
+                % inflation)
+
+        self.inflation = inflation
+        self.k_extra = self.k_inflate
+        self.exog_names.insert(0, 'inflate_const')
```

Comment:
this hardcodes const at 0 position

Reply:
Yes, this line hardcode const at 0 position, but in the next line i insert `inflation_x_i` into 0 position too, as a result,  const parameter will be after all `inflate_x_i` in summary.